src/Microsoft.Crank.AzureDevOpsWorker/Program.cs

```diff
@@ -48,15 +57,64 @@ public static int Main(string[] args)
                     queue = Environment.GetEnvironmentVariable(queue);
                 }
 
-                await ProcessAzureQueue(connectionString, queue);
+                if (certBasedAuth.HasValue() && !(certClientId.HasValue() && certTenantId.HasValue() && certThumbprint.HasValue()))
+                {
+                    Console.WriteLine("If using cert based auth, must provide client id, tenant id, and thumbprint.");
+                    return;
+                }
+
+                if(certBasedAuth.HasValue())
+                {
+                    await ProcessAzureQueue(connectionString, queue, certBasedAuth.HasValue(), certPath.Value(), certThumbprint.Value(), certTenantId.Value(), certClientId.Value());
+                }
+                else
+                {
+                    await ProcessAzureQueue(connectionString, queue);
+                }
             });
 
             return app.Execute(args);
         }
 
-        private static async Task ProcessAzureQueue(string connectionString, string queue)
+        private static async Task ProcessAzureQueue(string connectionString, string queue, bool certBasedAuth = false, string certPath = "", string certThumbprint = "", string certTenantId = "", string certClientId = "")
```

Comment:
Maybe we should create a reusable type that puts all these certificate related options into a single class and then we just need to pass around that object instead, should make things a little cleaner. This would eliminate the need for a boolean to indicate if certBasedAuth should be used since we can just use `null` to represent that.

Reply:
This likely makes the most sense.